<?xml version="1.0"?>
<launch>
  <arg name="robot_name" default="nexus0" />
  <arg name="pose_x" default="0" />
  <arg name="pose_y" default="0" />
  <arg name="yaw" default="0" />
  <arg name="goal_x" default="0" />
  <arg name="goal_y" default="0" />
  <arg name="kp" default="0" />
  <arg name="swarm_size" default="2" />
  <arg name="robot_id" default="1" />
  


  <group ns="$(arg robot_name)">
    <param name="tf_prefix" value="$(arg robot_name)" />
    <!-- Load robot description -->
    <include file="$(find nexus_4wd_mecanum_description)/launch/nexus_4wd_mecanum_description.launch" />

    <!-- Spawn the robot -->
    <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model"
          args="-urdf -model $(arg robot_name) -param robot_description -x $(arg pose_x) -y $(arg pose_y) -z 0.5 -Y $(arg yaw)" />
    
    <!-- Start publishing robot goal point -->
    <node name="pub_goal" pkg="cbf_simulation" type="pub_goal.py" clear_params="true"/>
        <param name="robot_name" value="$(arg robot_name)" /> 
        <param name="x" value="$(arg goal_x)" /> 
        <param name="y" value="$(arg goal_y)" /> 


    <node name="pub_tf" pkg="cbf_simulation" type="pub_tf.py" clear_params="true"/>
        <param name="robot_name" value="$(arg robot_name)" />    
        <param name="pose_x" value="$(arg pose_x)" /> 
        <param name="pose_y" value="$(arg pose_y)" /> 
        <param name="pose_yaw" value="$(arg yaw)" />


    <node name="odom_tf" pkg="cbf_simulation" type="odom_tf.py" clear_params="true"/>
        <param name="robot_name" value="$(arg robot_name)" />    

    <!-- Controller -->
    <node name="controller" pkg="cbf_simulation" type="controller.py" output="screen"  clear_params="true"/>
        <param name="robot_name" value="$(arg robot_name)" /> 
        <param name="gx" value="$(arg goal_x)" /> 
        <param name="gy" value="$(arg goal_y)" /> 
        <param name="kp" value="$(arg kp)" /> 
        <param name="robot_id" value="$(arg robot_id)" /> 
        <param name="swarm_size" value="$(arg swarm_size)" /> 


  </group>
</launch>
